### PR TITLE
use .helmignore when identifying changed charts

### LINKED ISSUE #530: Feature: ignore changes in files like README.md?
**Is this a request for help?**: No.

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): FEATURE REQUEST

I am using the helm-chart-testing action, that uses the chart-testing tool (obviously :-) ). It complains, if I make small changes to e.g. a chart's README.md file, and do not bump the chart's version.

For things like README.md or similar it would be nice, if those files could be ignored by `ct list-changed`.

I found no commandline argument that would allow me to do that. I also found no hint if this is possible using the config file.

### PR DESCRIPTION
**Pull request by jouve**
**What this PR does / why we need it**:

use .helmignore when identifying changed charts.

The use of .helmignore is disabled by default (current behaviour) and enabled with `--use-helmignore`

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #530

**Special notes for your reviewer**:

the FilterFiles function is adapted from https://github.com/helm/helm/blob/main/pkg/chart/loader/directory.go#L46

### CODE REVIEW

**Comment by jouve** [comment]
Up

**Comment by jouve** [comment]
up

**Review by davidkarlsen — COMMENTED** [review]
sorry for the late response.
can you rebase so we get a new build and see if the docs step passes?

**Comment by davidkarlsen** [comment]
please regen the docs and check in and it will pass

File: pkg/ignore/ignore_test.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package ignore
```
Comment (cpanato) [inline pkg/ignore/ignore_test.go:1]: add the boilerplate header

**Review by cpanato — CHANGES_REQUESTED** [review]
seems ok so far, but did not look in depth, i need a bit more time

**Review by cpanato — APPROVED** [review]

**Comment by mnonnenmacher** [comment]
@cpanato Could you please make a release with this feature? It's confusing that the `--use-helmignore` option is mentioned in the docs but not available in the latest version.

**Comment by cpanato** [comment]
@mnonnenmacher released https://github.com/helm/chart-testing/releases/tag/v3.12.0

**Comment by mnonnenmacher** [comment]
@cpanato Great, thank you!

**Pull request merged**